### PR TITLE
Validate Json for filter

### PR DESCRIPTION
**Pull request by mattcui**
Added common function `ValidateJson` in `common/utility.go`, and added unit test code to cover true/false path where filter is a valid Json string or not.

### CODE REVIEW

File: services/softlayer_account_test.go [hunk 1]
```diff
@@ -129,10 +129,15 @@ var _ = Describe("SoftLayer_Account_Service", func() {
 		})
 
 		It("returns an array of datatypes.SoftLayer_Virtual_Disk_Image", func() {
-			virtualDiskImages, err := accountService.GetVirtualDiskImagesWithFilter("fake-filter")
+			virtualDiskImages, err := accountService.GetVirtualDiskImagesWithFilter(`{"correct-filter":"whatever"}`)
 			Expect(err).ToNot(HaveOccurred())
 			Expect(virtualDiskImages).ToNot(BeNil())
 		})
+
+		It("returns an error due to failed Josn validation", func() {
```
Comment (maximilien) [inline services/softlayer_account_test.go]: Little typo here in description, should be `Json`

File: services/softlayer_account_test.go [hunk 2]
```diff
@@ -161,6 +166,24 @@ var _ = Describe("SoftLayer_Account_Service", func() {
 		})
 	})
 
+	Context("#GetBlockDeviceTemplateGroupsWithFilter", func() {
+		BeforeEach(func() {
+			fakeClient.DoRawHttpRequestResponse, err = common.ReadJsonTestFixtures("services", "SoftLayer_Account_Service_getBlockDeviceTemplateGroups.json")
```
Comment (maximilien) [inline services/softlayer_account_test.go]: No need for new line after `BeforeEach`

File: common/utility.go [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package common
+
+import (
+	"encoding/json"
+)
+
+func ValidateJson(s string) (bool, error) {
```
Comment (maximilien) [inline common/utility.go]: So I expected a new test file `utility_test.go` with test for the new `ValidateJson` function

**Comment by maximilien** [comment]
Hi, Matt, 

Overall good idea and like it. However, missing direct tests for new method and a few minor issues.

Please address and let me know and I will merge. Thanks,

@maximilien

File: services/softlayer_account_test.go (hunk 1 above)
Comment (mattcui) [inline services/softlayer_account_test.go]: Sorry, corrected it.

File: services/softlayer_account_test.go (hunk 2 above)
Comment (mattcui) [inline services/softlayer_account_test.go]: Do you mean there should be no empty line between 2 "It"?

```go
	Context("#GetBlockDeviceTemplateGroupsWithFilter", func() {
		BeforeEach(func() {
			fakeClient.DoRawHttpRequestResponse, err = common.ReadJsonTestFixtures("services", "SoftLayer_Account_Service_getBlockDeviceTemplateGroups.json")
			Expect(err).ToNot(HaveOccurred())
		})
		
		It("returns an array of datatypes.SoftLayer_Virtual_Guest_Block_Device_Template_Group", func() {
			groups, err := accountService.GetBlockDeviceTemplateGroupsWithFilter(`{"correct-filter":"whatever"}`)
			Expect(err).ToNot(HaveOccurred())
			Expect(groups).ToNot(BeNil())
		})
		It("returns an error due to failed Json validation", func() {
			_, err := accountService.GetBlockDeviceTemplateGroupsWithFilter(`{{"wrong-filter":"whatever"}`)
			Expect(err).To(HaveOccurred())
		})
	})
```

File: common/utility.go (hunk 3 above)
Comment (mattcui) [inline common/utility.go]: Added it, please review. Thanks.

**Comment by mattcui** [comment]
@maximilien, I finished all changes, please review and merge if there is no problem. Thanks.

**Comment by maximilien** [comment]
As I said in my email, please separate Godeps changes to different PRs to commits that are changes in the code, unless you made changes in the dependent code. 

I am therefore closing this PR and resubmit as per email. Thanks.

**Pull request closed without merging**